src/shared/copy.c

```diff
@@ -798,6 +798,10 @@ static int fd_copy_regular(
         (void) futimens(fdt, (struct timespec[]) { st->st_atim, st->st_mtim });
         (void) copy_xattr(fdf, NULL, fdt, NULL, copy_flags);
 
+        r = fd_verify_regular_full(fdf, /* check_removed = */ true);
+        if (r < 0)
+                return r;
```

Comment:
so i think this test only makes sense if used in conjunction with that FALLOC_FL_PUNCH_HOLE delete pattern we have. In the usual case I see no reason why we'd want to add this test, because usually when files are deleted they aren't weirdly destroyed like we do it for the journal vacuuming...

Hence I think this should be hidden behind a new flag, i.e. COPY_VERIFY_LINKED or so? and then only set that in the journal code.